src/Arcus.Scripting.IntegrationAccount/Scripts/Set-AzIntegrationAccountAssemblies.ps1

```diff
@@ -0,0 +1,73 @@
+    [parameter(Mandatory = $false)][string] $AssembliesFolder = $(if ($AssemblyFilePath -eq '') { throw "Either the file path of a specific assembly or the file path of a folder containing multiple assemblies is required, e.g.: -AssemblyFilePath 'C:\Assemblies\assembly.dll' or -AssembliesFolder 'C:\Assemblies'" }),
+    [Parameter(Mandatory = $false)][string] $ArtifactsPrefix = ''
+)
+
+if ($AssemblyFilePath -ne '' -and $AssembliesFolder -ne '') {
+    throw "Either the file path of a specific assembly or the file path of a folder containing multiple assemblies is required, e.g.: -AssemblyFilePath 'C:\Assemblies\assembly.dll' or -AssembliesFolder 'C:\Assemblies'"
+}
+
+function UploadAssembly {
+    param
+    (
+        [System.IO.FileInfo][parameter(Mandatory = $true)]$Assembly
+    )
+
+    $assemblyName = $Assembly.BaseName
+    if ($ArtifactsPrefix -ne '') {
+        $assemblyName = $ArtifactsPrefix + $assemblyName
+    }
+    Write-Host "Uploading assembly '$assemblyName' into the Integration Account '$Name'."
+
+    $existingAssembly = $null
+    try {
+        Write-Verbose "Checking if the assembly '$assemblyName' already exists in the Integration Account '$Name'."
+        $existingAssembly = Get-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -ErrorAction Stop
+    }
+    catch {
+        if ($_.Exception.Message.Contains('could not be found')) {
+            Write-Verbose "No assembly '$assemblyName' could not be found in Azure Integration Account '$Name'."
+        }
+        else {
+            throw $_.Exception
+        }
+    }
+        
+    try {
+        if ($existingAssembly -eq $null) {
+            Write-Verbose "Creating assembly '$assemblyName' in Azure Integration Account '$Name'"
+            $createdAssembly = New-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -AssemblyFilePath $Assembly.FullName -ErrorAction Stop
+            Write-Verbose ($createdAssembly | Format-List -Force | Out-String)
+        }
+        else {
+            Write-Verbose "Updating assembly '$assemblyName' in Azure Integration Account '$Name'"
+            $updatedAssembly = Set-AzIntegrationAccountAssembly -ResourceGroupName $ResourceGroupName -IntegrationAccount $Name -Name $assemblyName -AssemblyFilePath $Assembly.FullName -ErrorAction Stop
+            Write-Verbose ($updatedAssembly | Format-List -Force | Out-String)
+        }
+        Write-Host "Assembly '$assemblyName' has been uploaded into the Azure Integration Account '$Name'."
+    }
+    catch {
+        Write-Error "Failed to upload assembly '$assemblyName' in Azure Integration Account '$Name': '$($_.Exception.Message)_'"
```

Comment:
Why do we write to the error stream here, and on line 35, we throw?

Reply:
Again, this behavior is copied from the already existing scripts to upload schemas and maps 😉 
On line 35 we throw since we need to terminate the script there because if that statement fails we don't know if we need to do a create or an update.

Reply:
Correct, if the block that is executed before line 35 fails, there is no point in continuing so processing should be stopped. Basically, if the error is anything else than referring to the fact that the assembly in this case, cannot be found, it means there is some kind of different issue going on with the Integration account, so no need to continue with anything else as all other following actions will fail as well.
While at this point, simply logging the error and making it visible to the end-user is sufficient.

Basically, what Pim said ;)

Reply:
Ok, makes sense! Thx for clarifying.